test/main/utils/type_operator_test.dart

```diff
@@ -0,0 +1,88 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:tmail_ui_user/main/exceptions/remote_exception.dart';
+
+bool isNotBadCredentialsExceptionUseIsNotOperator(dynamic exception) {
+  return exception is! BadCredentialsException;
+}
+
+bool isBadCredentialsExceptionUseIsOperator(dynamic exception) {
+  return exception is BadCredentialsException;
+}
+
+bool isBadCredentialsExceptionUseNotIsOperator(dynamic exception) {
+  return exception !is BadCredentialsException;
+}
+
+void main() {
+  group('is! operator unit test', () {
+    test('should return true when exception is not a BadCredentialsException', () {
+      // Arrange
+      const connectionError = ConnectionError();
+
+      // Act
+      final result = isNotBadCredentialsExceptionUseIsNotOperator(connectionError);
+
+      // Assert
+      expect(result, true);
+    });
+
+    test('should return false when exception is a BadCredentialsException', () {
+      // Arrange
+      const badCredentialsException = BadCredentialsException();
+
+      // Act
+      final result = isNotBadCredentialsExceptionUseIsNotOperator(badCredentialsException);
+
+      // Assert
+      expect(result, false);
+    });
+  });
+
+  group('is operator unit test', () {
+    test('should return true when exception is a BadCredentialsException', () {
+      // Arrange
+      const badCredentialsException = BadCredentialsException();
+
+      // Act
+      final result = isBadCredentialsExceptionUseIsOperator(badCredentialsException);
+
+      // Assert
+      expect(result, true);
+    });
+
+    test('should return false when exception is not a BadCredentialsException', () {
+      // Arrange
+      const connectionError = ConnectionError();
+
+      // Act
+      final result = isBadCredentialsExceptionUseIsOperator(connectionError);
+
+      // Assert
+      expect(result, false);
+    });
+  });
+
+  group('!is operator unit test', () {
+    test('should return true when exception is a BadCredentialsException', () {
+      // Arrange
+      const badCredentialsException = BadCredentialsException();
+
+      // Act
+      final result = isBadCredentialsExceptionUseIsOperator(badCredentialsException);
```

Comment:
I assume you wanted to use the function `isBadCredentialsExceptionUseNotIsOperator` here. Also on the line 82

Reply:
Oh, that was my mistake. Fixed.